dspace-api/src/main/java/org/dspace/access/status/DefaultAccessStatusHelper.java

```diff
@@ -156,4 +162,83 @@ private String caculateAccessStatusForDso(Context context, DSpaceObject dso, Dat
         }
         return RESTRICTED;
     }
+
+    /**
+     * Look at the policies of the primary (or first) bitstream of the item to retrieve its embargo.
+     *
+     * If the item is null, simply returns an empty map with no embargo information.
+     *
+     * @param context     the DSpace context
+     * @param item        the item to embargo
+     * @return an access status value
+     */
+    @Override
+    public String getEmbargoFromItem(Context context, Item item)
+            throws SQLException {
+        Date embargoDate;
+
+        if (item == null) {
+            return null;
+        }
+        // Consider only the original bundles.
+        List<Bundle> bundles = item.getBundles(Constants.DEFAULT_BUNDLE_NAME);
+        // Check for primary bitstreams first.
+        Bitstream bitstream = bundles.stream()
+                .map(bundle -> bundle.getPrimaryBitstream())
+                .filter(Objects::nonNull)
+                .findFirst()
+                .orElse(null);
+        if (bitstream == null) {
+            // If there is no primary bitstream,
+            // take the first bitstream in the bundles.
+            bitstream = bundles.stream()
+                    .map(bundle -> bundle.getBitstreams())
+                    .flatMap(List::stream)
+                    .findFirst()
+                    .orElse(null);
+        }
+
+        if (bitstream == null) {
+            return null;
+        }
+
+        embargoDate = this.retrieveLongestEmbargo(context, bitstream);
+
+        return embargoDate != null ? embargoDate.toString() : null;
+    }
+
+    /**
+     *
+     */
+    private Date retrieveLongestEmbargo(Context context, Bitstream bitstream) throws SQLException {
```

Comment:
To better align with how embargoes work, I think we need to change this to `retrieveShortestEmbargo`.

If somehow you have two Anonymous policies with different embargo dates (which should be very rare), once the **earliest embargo date passes**, that Item will be publicly accessible.  

So, in this scenario, if we return the *longest Embargo*, I believe that would result in an incorrect embargo date.

Reply:
Done now